data_processors/pipeline/orchestration/tests/test_tumor_normal_step.py

```diff
@@ -201,6 +299,45 @@ def test_prepare_tumor_normal_jobs_issue_475_skip(self):
 
         self.assertEqual(len(job_list), 0)
 
+    def test_prepare_tumor_normal_jobs_research_tumor_research_normal(self):
+        """
+        python manage.py test data_processors.pipeline.orchestration.tests.test_tumor_normal_step.TumorNormalStepUnitTests.test_prepare_tumor_normal_jobs_research_tumor_research_normal
+        :return:
+        """
+        # We expect a tn job to be created in this test
+        build_tn_mock_research_tumor_clinical_normal()
+
+        mock_lbr_normal: LibraryRun = LibraryRunFactory()
+        mock_lbr_tumor: LibraryRun = TumorLibraryRunFactory()
+
+        meta_list = LabMetadata.objects.all()
+
+        job_list, subjects, submitting_subjects = tumor_normal_step.prepare_tumor_normal_jobs(meta_list)
+
+        logger.info(f"job_list: {json.dumps(job_list)}")
+
+        job_dict = job_list[0]
+        self.assertEqual(job_dict['subject_id'], tn_mock_subject_id)
+        self.assertEqual(job_dict['fastq_list_rows'][0]['rglb'], TestConstant.library_id_normal.value)
+        self.assertEqual(job_dict['fastq_list_rows'][0]['read_1']['location'], tn_mock_normal_read_1)
+
+    def test_prepare_tumor_normal_jobs_clinical_tumor_research_normal(self):
+        """
+        python manage.py test data_processors.pipeline.orchestration.tests.test_tumor_normal_step.TumorNormalStepUnitTests.test_prepare_tumor_normal_jobs_clinical_tumor_research_normal
+        :return:
+        """
+        # We do NOT expect a tn job to be created in this test
+        build_tn_mock_clinical_tumor_research_normal()
+
+        mock_lbr_normal: LibraryRun = LibraryRunFactory()
+        mock_lbr_tumor: LibraryRun = TumorLibraryRunFactory()
+
+        meta_list = LabMetadata.objects.all()
+
+        job_list, subjects, submitting_subjects = tumor_normal_step.prepare_tumor_normal_jobs(meta_list)
+
+        logger.info(f"job_list: {json.dumps(job_list)}")
+
```

Comment:
Great stuff! Thanks for unit testing. 👍

Reply:
Ah. One thing. Could you add assertion pls. Like, `len(job_list) == 0` for example.

Reply:
Resolved in f59e06500b0f225a4a86a55e596e8666d204f0c9